File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/core/network/prnumber/IPRNumberDefaultRequest.java
package com.pplus.prnumberbiz.core.network.prnumber;

import com.pplus.prnumberbiz.core.network.model.response.NewResultResponse;

import network.common.PplusCallback;
import retrofit2.Call;

/**
 * Created by j2n on 2016. 9. 12..
 */
public interface IPRNumberDefaultRequest<T> extends IPRNumberRequest<T>{

    IPRNumberDefaultRequest<T> setCallback(PplusCallback<NewResultResponse<T>> callback);

    IPRNumberDefaultRequest<T> setTag(Object tag);

    IPRNumberDefaultRequest setRequestCall(Call<NewResultResponse<T>> requestCall);

    PRNumberRequest<T> build();
}

File: LuckyBol/app/src/main/java/com/pplus/luckybol/core/network/prnumber/IPRNumberDefaultRequest.kt
package com.pplus.luckybol.core.network.prnumber

import com.pplus.networks.common.PplusCallback
import com.pplus.luckybol.core.network.model.response.NewResultResponse
import retrofit2.Call

/**
 * Created by j2n on 2016. 9. 12..
 */
interface IPRNumberDefaultRequest<T> : IPRNumberRequest<T> {
    override fun setCallback(callback: PplusCallback<NewResultResponse<T>>): IPRNumberDefaultRequest<T>
    override fun setTag(tag: Any): IPRNumberDefaultRequest<T>
    fun setRequestCall(requestCall: Call<NewResultResponse<T>>): IPRNumberDefaultRequest<T>
    override fun build(): PRNumberRequest<T>
}